### PR TITLE
Initial proposal for Population Variance

### PR DESCRIPTION
**Pull request by fdbeirao**
This is a proposed implementation for #36

I am still a bit confused with the edge case that on main, if I provide `[]` instead of a List with numbers, I am not getting the expected `Err NotEnoughItems` and I don't understand why.

Also, trying to run `roc test` (after uncommenting the `expect`) crashes, as reported on https://github.com/roc-lang/roc/issues/5482

---

Current output of running `roc run --linker=legacy`

`σ² = 5032.13888888889`

### CODE REVIEW

File: ci_scripts/expect_scripts/SafeMath.exp [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# uncomment line below for debugging
+# exp_internal 1
+
+set timeout 7
+
+spawn ./roc_nightly/roc run ./examples/SafeMath/main.roc
+
+
+expect "147.66666666666666\r\n" {
```
Comment (fdbeirao) [inline ci_scripts/expect_scripts/SafeMath.exp:11]: The current implementation outputs `σ² = 147.66666666666666\r\n`

(tested with `roc run --linker=legacy` and `roc --version` == `roc nightly pre-release, built from commit ef05ec6 on Di 30 Mai 2023 09:34:00 UTC`)

**Comment by fdbeirao** [comment]
Thank you for your commits @Anton-4, and for adding the output expectation tests. 🙏 
I am happy I could make a small contribution, roc is a delightful language.

File: ci_scripts/expect_scripts/SafeMath.exp (hunk 1 above)
Comment (Anton-4) [reply ci_scripts/expect_scripts/SafeMath.exp]: Good catch :)

**Comment by lukewilliamboswell** [comment]
This doesn't look blocked any longer. 

@fdbeirao are you interested in resolving up the merge conflicts and updating to latest main?

**Review by lukewilliamboswell — APPROVED** [review]
Thank you @fdbeirao this is good, I made some changes to get it working now it's not blocked.

**Review by Anton-4 — APPROVED** [review]
